src/generic/trilinos_eigen_solver.h

```diff
@@ -529,19 +542,23 @@ namespace oomph
       {
         Linear_solver_pt->enable_resolve();
       }
-      // Solve the first vector
 
-      DoubleVector X(x.distribution_pt());
+      // Solve for the first vector (no need for a distribution)
```

Comment:
I recall discussions with Richard about efficiency of setting up the distributions, whether it was better to assign the distribution first or not. I can't now recall the details, but that was why the distributions were assigned here. Changing distributions will (obviously) involve some MPI communication. If everything does get set up automatically and efficiently then (obviously) that's fine.

Reply:
Setting up DoubleVectors (especially in a serial context) always scares me because I can never remember how the distribution thing works (or why I would need to specify it in a serial driver code). Answer: You don't have to and things will be assigned automatically (and generally consistently, using a uniform distribution over all processors). So you can't go wrong with not providing a distribution. Setting them up does not require communication since the distribution is "obvious" once you know how many processors you're running on.

  Done.